vetiver/helpers.py

```diff
@@ -0,0 +1,51 @@
+from functools import singledispatch
+import pandas as pd
+import pydantic
+
+
+@singledispatch
+def api_data_to_frame(pred_data) -> pd.DataFrame:
+    """Convert prototype to dataframe data
+
+    Parameters
+    ----------
+    pred_data : pydantic.BaseModel
+        User data from given to API endpoint
+
+    Returns
+    -------
+    pd.DataFrame
+        BaseModel data translated into DataFrame
+    """
+
+    raise TypeError("Data should be list, pydantic.BaseModel, pd.DataFrame")
+
+
+@api_data_to_frame.register(pydantic.BaseModel)
+@api_data_to_frame.register(list)
+def _(pred_data):
+
+    return pd.DataFrame([dict(s) for s in pred_data])
+
+
+@api_data_to_frame.register(dict)
+def _dict(pred_data):
+    return api_data_to_frame([pred_data])
+
+
+# possible other names
```

Comment:
still trying to figure out the best name for this function, here's a list of a few options

Reply:
I don't know if any of this helps, but in R:

- The function that converts a dataframe to a new dataframe using the `prototype` is [`vetiver_type_convert()`](https://github.com/rstudio/vetiver-r/blob/06f3a15928907206d4c8d4932b9e03e3d7309468/R/handlers.R#L76).
- The conversion from dataframe to JSON, then into the API, then from JSON output back to dataframe all happens [inside the `predict` method for a `vetiver_endpoint()`](https://github.com/rstudio/vetiver-r/blob/06f3a15928907206d4c8d4932b9e03e3d7309468/R/endpoint.R#L20). By the time the dataframe gets here, it has already gone through `vetiver_type_convert()`.

Reply:
This is useful! Ah, the `predict` method in Python does the same user data -> JSON -> POST to API ... get JSON back from API -> convert to dataframe. 

This function is more like `vetiver_type_convert`, as it is using `prototype`. It is happening inside the API, so it might be nice to have that in the function name as well. I do like the idea of using the verb "convert," since that is what is happening here.